client/app/vps/upgrade/vps-upgrade.controller.js

```diff
@@ -0,0 +1,85 @@
+class VpsUpgradeCtrl {
+    constructor ($filter, $translate, $q, VpsService) {
+        this.$filter = $filter;
+        this.$translate = $translate;
+        this.$q = $q;
+        this.Vps = VpsService;
+
+        this.loaders = {
+            init: false
+        };
+
+        this.step = 0;
+        this.order = null;
+        this.selectedModel = {};
+        this.upgradesList = null;
+        this.agree = {
+            value: false
+        };
+    }
+
+    $onInit () {
+        this.loaders.init = true;
+        this.loadNextStep();
+    }
+
+    canLoadNextStep (condition, step) {
+        if (condition) {
+            this.loadNextStep(step);
+        }
+    }
+
+    loadNextStep (step) {
+        this.step = step || this.step + 1;
```

Comment:
We should have named steps instead of an integer. An array of steps:
```
this.steps = [{
    id: "model",
    status: "completed"
}, {
    id: "contracts",
    status: "active"
}, {
    id: "summary",
    status: "disabled"
}];
```
This way it will be easier to add/remove/re-arrange steps in the future.

Reply:
I'm still not quite sure about having status in here, we'll need to talk about it.

Reply:
Great idea, changing this (btw it needs to be settled in several places)